plugins/output-mongodb/src/main/scala/com/stratio/sparkta/plugin/output/mongodb/MongoDbOutput.scala

```diff
@@ -51,25 +49,26 @@ class MongoDbOutput(keyName: String,
 
   override val retrySleep = properties.getString("retrySleep", DefaultRetrySleep).toInt
 
-  override val idAsField = Try(properties.getString("idAsField").toBoolean).getOrElse(false)
-
   override val textIndexFields = properties.getString("textIndexFields", None).map(_.split(FieldsSeparator))
 
   override val language = properties.getString("language", None)
 
-  override def setup: Unit =
-    if (bcSchema.isDefined) {
+  override def setup : Unit = {
+    val db = connectToDatabase
```

Comment:
The db client isn't optional, so I think i should be encapsulated in DAO. At this point, if DB is none, setup will not be correct and only we'll have a warning as log output.